### DISCUSSION TITLE
Why are Levenshtein distances not using the rapidfuzz implementation?

### DISCUSSION THREAD

**Discussion by zmbc** [duckdb/duckdb#15234] [Q&A] [answered]
I recently contributed #12079, which allows the DuckDB user to access a threshold argument in rapidfuzz's Jaro and Jaro-Winkler string distances. I was planning to do the same for Levenshtein and Damerau-Levenshtein, but it appears they are not using rapidfuzz ([Levenshtein](https://github.com/duckdb/duckdb/blob/07780a0d22a03ab0bfd058c1ecc1fa16dc86fd18/extension/core_functions/scalar/string/levenshtein.cpp), [Damerau-Levenshtein](https://github.com/duckdb/duckdb/blob/07780a0d22a03ab0bfd058c1ecc1fa16dc86fd18/extension/core_functions/scalar/string/damerau_levenshtein.cpp)). Before I open a PR to switch to the rapidfuzz implementation and expose the argument, I wanted to check that I'm not missing a reason that won't work/isn't desirable.

Perhaps @c5sire (https://github.com/duckdb/duckdb/pull/1441) or @ADBond (https://github.com/duckdb/duckdb/pull/7035) would know?

**Comment by ADBond** [comment, marked as answer]
No particular reason with Damerau-Levenshtein, I just implemented the algorithm from [this paper](https://doi.org/10.1145%2F321879.321880) to get something working, and wasn't aware rapidfuzz has a C++ implementation.

This is coincidental timing - had just started looking at adding a threshold to Levenshtein + some general improvements to speed up DL, so sounds like a good idea.
